lib/src/main/java/com/basistheory/android/view/CardNumberElement.kt

```diff
@@ -38,6 +42,8 @@ class CardNumberElement @JvmOverloads constructor(
         isValid: Boolean
     ): ChangeEvent {
         val cardResult = cardBrandEnricher.evaluateCard(getDigitsOnly(value))
+        cardDetails = cardResult.cardDetails
+
         val eventDetails = cardResult.cardDetails?.brand?.let { brand ->
```

Comment:
It seems like now that `cardDetails` is persisted at the class level, we shouldn't need to re-compute the `cardResult` here (we compute it 2x per change event - one in `beforeTextChanged` and another in `createElementChangeEvent`). The only hangup is that `complete` is computed in `cardResult` and not within `cardDetails`. I'm wondering whether we could/should move this computation out to something like `fun CardNumberElement.isComplete(): Boolean` which uses the `cardDetails` and the current value from `getText()`.